dex/networks/eth/params.go

```diff
@@ -130,7 +131,11 @@ func GweiToWei(v uint64) *big.Int {
 
 // GweiToWei converts *big.Int Wei to uint64 Gwei.
 func WeiToGwei(v *big.Int) uint64 {
-	return new(big.Int).Div(v, BigGweiFactor).Uint64()
+	vGwei := new(big.Int).Div(v, BigGweiFactor)
+	if vGwei.IsUint64() {
+		return vGwei.Uint64()
+	}
```

Comment:
We have a lot of redundancy with these helpers.

- server/asset/eth/common.go has `ToGwei` helpers that can be removed.  Also that one is bizarrely used in client (ExchangeWallet)
- dex/networks/eth/common.co has `ToGwei` and `ToWei`

Also, the docs on this function say `GweiToWei` instead of `WeiToGwei`.

Reply:
Two things I was trying to avoid with the ToGwei.

- A package level BigGweiFactor. This is mutable.  Is this safe? For one a consumer could accidentally mutate it. `big` isn't the easiest package to use imo. Whenever you have `a := b.Add(c, d )` you might not expect b to change here, because we just want to add c and d, but b is set to those and returned as a. Also, could an asset that is not one of ours be added that maliciously changes this? I'm not sure if that is something we are worried about, but imo starting with a constant, or at least not exporting it, is safer.

As an example, add this test to the top of `eth_test.go` and another test will fail:
```
func TestMutate(t *testing.T) {
	fmt.Println("**************")
	fmt.Println(dexeth.BigGweiFactor)
	c := big.NewInt(2)
	d := big.NewInt(3)
	a := dexeth.BigGweiFactor.Add(c, d)
	fmt.Printf("%p %p\n", a, dexeth.BigGweiFactor)
	fmt.Println(dexeth.BigGweiFactor)
}
```

- A `big.Int` may not fit into a `uint64`. It could be too large or negative. It would be unexpected, but have untested consequences. Should we at least panic?

Reply:
Good call on the `BigGweiFactor`. 

> * A `big.Int` may not fit into a `uint64`.

That's the purpose of both `ToGwei` and the new `IsUnit64` check in `WeiToGwei`. Which brings us back to @chappjc's original observation 

> We have a lot of redundancy with these helpers.

I have no problem having both `ToGwei` and `WeiToGwei`, the choice of which depends on if we want 1) to explicitly check for overflow/sign errors, or 2) just fall back to zero when not suitable for `uint64`. `WeiToGwei` clearly has some syntactic advantage, but whether it's appropriate needs to be considered in context.

I do think we should rename `ToGwei`.